### PR TITLE
Add support for <i18n> blocks of SFC.

### LINKED ISSUE #11: Support for single-file components
It seems like this plugin does not support localization with `<i18n>` tags inside single-file components (as [explained in the manual](https://kazupon.github.io/vue-i18n/guide/sfc.html)) - do you think you can add support for this?

### PR DESCRIPTION
**Pull request by ota-meshi**
This PR adds support for the `<i18n>` block.

close #11

I have made the following changes to support `<i18n>`.

- Change `@intlify/vue-i18n/no-html-messages` rule to also verify the json in the `<i18n>` block.
- Change `@intlify/vue-i18n/no-missing-keys` rule to not report if the message used by SFC is inside `<i18n>` block.
- Change `@intlify/vue-i18n/no-unused-keys` rule to report even if the message in the `<i18n>` block is not used by that SFC.

In addition, If you don't validate the `.json` and using `<i18n>`, the rule no longer reports an error even if you don't have `settings`.

Parsing the `<i18n>` block understands the following attributes:

- `src` ... The `@intlify/vue-i18n/no-missing-keys` rule also looks for keys in the file specified by `src`.
- `locale` ... If this is specified, it is determined that JSON does not include locale, similar to `vue-i18n-loader`.

### CODE REVIEW

**Comment by kazupon** [comment]
Thank you for your PR!
Awesome !

I'll try to review, please wait! 🙏

File: lib/utils/collect-keys.js [hunk 1]
```diff
@@ -37,96 +48,123 @@ function getKeyFromVAttribute (node) {
...
+  }
+
+  if (node.templateBody) {
+    vueESLintParser.AST.traverseNodes(node.templateBody, {
+      enterNode,
+      leaveNode () {}
+    })
+  }
+  vueESLintParser.AST.traverseNodes(node, {
+    visitorKeys: visitorKeys,
```
Comment (kazupon) [inline lib/utils/collect-keys.js:159]: You can use `shorthand prop name` sytanx.
```js
vueESLintParser.AST.traverseNodes(node, {
    visitorKeys
```

**Comment by kazupon** [comment]
@sosukesuzuki 
Thanks for your information!

Do mysticatea know about that issue?
I've just checked it out vue-eslint-parser github issues, but I couldn't find it. 😞 
You might want to register that issue in the github issues. :)

**Comment by ota-meshi** [comment]
I didn't know that vue-template-compiler could parse a block without HTML escaping😅
But when I check it with [astexplorer.net](https://astexplorer.net/), `@vue/compiler-core` seems to need escaping. 
Hmm... I don't know which is right.

[demo](https://astexplorer.net/#/gist/5de65cc265946ec89349544d7afa8816/8be1016df1904518aee8883cb7546586621fc2ac)

File: lib/utils/collect-keys.js (hunk 1 above)
Comment (ota-meshi) [reply lib/utils/collect-keys.js]: I fixed this. I also enabled the "object-shorthand" rule in `.eslintrc`.

**Comment by ota-meshi** [comment]
@kazupon Thank you for reviewing!
I changed this PR. so check again it.

**Comment by wolfgangwalther** [comment]
Does this also work with `<i18n lang="yaml" ...>` blocks?

**Comment by ota-meshi** [comment]
yaml is not yet supported.

**Comment by tmcdos** [comment]
> I didn't know that vue-template-compiler could parse a custom block without HTML escaping
> But when I check it with [astexplorer.net](https://astexplorer.net/), `@vue/compiler-core` seems to need escaping.
> Hmm... I don't know which is right.
> 

Is it possible to use the ability of Vue-Loader for custom blocks ? https://vue-loader.vuejs.org/guide/custom-blocks.html

**Pull request merged**